README.md

```diff
@@ -85,3 +85,34 @@ This configuration will only send multicast packets to the interface assigned
 the `192.168.1.1` address. The interface lookup is dynamic each time an mDNS
 query is sent, so if the address moves to a different interface the plugin
 will automatically switch to the new one.
+
+## Service Discovery
+
+Queries to mDNS are constrained to the `local` domain, alternative domains that an mDNS server publishes are not supported by this plugin. This does not restrict you to `.local` TLD, as the address is referenced from the service responses belonging to the `local` domain.
```

Comment:
With #66 will this still be true? I know I had issues using anything other than .local as the mdns domain, which is why I hard-coded it in the plugin, but with that change in theory it should be possible to use others.

Reply:
For those conforming to mDNS spec, afaik `.local` is the only officially reserved TLD for mDNS. Although, those clients would not need to go through this CoreDNS plugin if they worked fine with regular mDNS :)

I can't say that I've looked too deeply into the domains part. I know that Avahi can publish to different domains to group/categorize/isolate your mDNS hosts, but the default is `local`. 

#66 appears to be about support for alternative TLDs other than `.local` which afaik is unsupported by typical mDNS clients (`nss-mdns` on Linux for the system resolver will only accept `.local` for mDNS queries with it's default minimal config in `nsswitch.conf`, but has a variant for supporting additional alternative TLDs, otherwise the query goes through regular DNS where `coredns-mdns` could respond, that also includes queries using the `.local` TLD that fail to meet the additional criteria for qualifying as an mDNS FQDN).

So uhh, yes this document should still be accurate even with #66 , you would need a way to specify other mDNS *domains* to search. Perhaps I could word this better as it can be a bit confusing? (easy to accidentally think it's referring to the same thing with `local` domain and `.local` TLD)

---

`avahi-publish -s helloworld _workstation._tcp 1234 --domain=local --host=hello.hostname.test`

With the above, I have:

- Published a service called `helloworld` 
- Under the `_workstation._tcp` service that this plugin scans for
- With a port of `1234` (irrelevant here but required for the command to work)
- Under the domain group `local` which the plugin scans for
- With the hostname of `hello.hostname.test`, the plugin currently  truncates length of the domain scanned `local` (5 chars + 1 more with `.`, resulting in `hello.hostnam`), and then appends the TLD used in the Corefile (in the bug report this was `mdnsbug`), resulting in `hello.hostnam.mdnsbug` as an `mdnsHost` the plugin will respond to a query for.

Thus `hello.hostnam.mdnsbug` maps to the discovered `helloworld` service with the hostname value `hello.hostname.test`(or it's IP address, not entirely sure how the plugin is working internally) to respond with the correct IP when queried. The problem was that it responds to the wrong request of `hello.hostnam.mdnsbug` instead of `hello.hostname.mdnsbug`.

---

Summary:

- `.local` and `local` are the standard TLD and domain.
- It's possible to publish an mDNS hostname to a different TLD and that TLD does not have to match a domain of the same name. When scanning for services (which this plugin does for discovery), the service publishes itself under a domain, and the TLD is part of the service metadata for it's hostname value (if one is provided).
- The plugin presently only searches `local` domain with two service types (`_workstation._tcp` is really the only one that matters here), it adds those to it's discovered `mdnsHosts` array and can lookup the hostname value to respond with, but the present bug is that the hostname value used to match with the Corefile TLD truncates `.local` from the end of the hostname value, such that if the TLD used differs in length, the intended hostname to match to is invalid.